esrfconcert/devices/motors/micos.py

```diff
@@ -172,3 +207,22 @@ async def _set_velocity(self, velocity):
     async def _home(self):
         await self._connection.execute('{} RefMove {}'.format(self._controller, self._index + 1))
         await self['state'].wait('standby', sleep_time=self._connection.sleep_between)
+
+
+class LaminoScanningMotor(ContinuousRotationMotor):
+
+    """An implementation with specific functionality for the rotary state of LAMINO-I"""
+
+    def __init__(self, controller, index, host, port, pusher1, pusher2):
+        super(RotationMotor, self).__init__()
+        _Base.__init__(self, controller, index, host, port)
+        self.pusher1 = pusher1
+        self.pusher2 = pusher2
+
+    async def _set_position(self, position):
+
+        if self.pusher1.is_pusher_out() == "Pusher out" and self.pusher2.is_pusher_out() == "Pusher out":
+            position = position.to(q.deg).magnitude
+            await self._set_position_in_steps(position)
+        else:
+            print("Pushers are not out!")
```

Comment:
Here you should raise an exception, just make a new class like this:
```python
class LaminoRotException(Exception):
    pass
```
and then `raise LaminoRotException(Pushers are not out)`. Of course the class name is up to you.